src/desktop.c

```diff
@@ -2298,13 +2330,20 @@ static void update_background(FmDesktop* desktop, int is_it)
             }
         }
 #if GTK_CHECK_VERSION(3, 0, 0)
-        xdisplay = GDK_WINDOW_XDISPLAY(root);
-        /* this code is taken from libgnome-desktop */
-        xpixmap = XCreatePixmap(xdisplay, RootWindow(xdisplay, screen_num),
-                                dest_w, dest_h, DefaultDepth(xdisplay, screen_num));
-        cache->bg = cairo_xlib_surface_create(xdisplay, xpixmap,
-                                              GDK_VISUAL_XVISUAL(gdk_screen_get_system_visual(screen)),
-                                              dest_w, dest_h);
+        if(IS_X11())
+        {
+#ifdef HAVE_X11
+            xdisplay = GDK_WINDOW_XDISPLAY(root);
+            /* this code is taken from libgnome-desktop */
+            xpixmap = XCreatePixmap(xdisplay, RootWindow(xdisplay, screen_num),
+                                    dest_w, dest_h, DefaultDepth(xdisplay, screen_num));
+            cache->bg = cairo_xlib_surface_create(xdisplay, xpixmap,
+                                                  GDK_VISUAL_XVISUAL(gdk_screen_get_system_visual(screen)),
+                                                  dest_w, dest_h);
+#endif
+        }
+        else
```

Comment:
This `else` does not make sense if no wayland enabled, it newer happens in that case because `IS_X11` will be `TRUE`.

Reply:
Yes, and the compiler will the optimize it out and #ifdef HAVE_WAYLAND isn't needed, because it's not some Wayland-specific API

Reply:
I wouldn't be so sure about compiler optimization. And even if that is the case then using `#if` improves the code readability not enforcing the reader to make all the logic calculations, that really helps to not spend extra time on that.